### PR TITLE
feat: added trading comp tables

### PR DESCRIPTION
**Pull request by glo12345**
Moving from services to contracts

### CODE REVIEW

File: database/102-up-timescale/20230802004951_trading_comp_init_epoch.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  id SERIAL NOT NULL,
+
+  -- start timestamp of epoch
+  start TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+
+  -- end timestamp of epoch
+  end TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+
+  -- application to be filtered for. Can be "none" (For Fluidity) or NULL (For All)
+  application ethereum_application
```
Comment (eli-d) [inline database/102-up-timescale/20230802004951_trading_comp_init_epoch.sql]: Do we know that an epoch has only one application?

**Review by eli-d — APPROVED** [review]

File: database/102-up-timescale/20230802004951_trading_comp_init_epoch.sql (hunk 1 above)
Comment (glo12345) [reply database/102-up-timescale/20230802004951_trading_comp_init_epoch.sql]: good call - they can track any number of apps

File: cmd/microservice-epoch-rotation/main.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+const DEFAULT_UNIX = -62135596800
+
+const (
+	// EnvAwsClusterName to determine which cluster to restart the services in
+	EnvRandomOrgSecret = `FLU_RANDOM_ORG_SECRET`
+
+	// EnvAwsServiceName to match services that include it in their names
+	EnvMeanEpochDays = `FLU_REWARD_EPOCH_MEAN_DAYS`
+
+	EnvEpochApplications = `FLU_REWARD_EPOCH_APPLICATIONS`
```
Comment (eli-d) [inline cmd/microservice-epoch-rotation/main.go]: comments don't match the variables

File: cmd/microservice-epoch-rotation/main.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+	} else {
+		currentEpochEndTime = currRewardEpoch.EndTime
+	}
+
+	log.Debug(func(k *log.Log) {
+		k.Format("currTime: %v", currentEpochEndTime)
+	})
+
+	// get next epoch length
+	gaussianRes := getGaussanValues(randomOrgSecret, 5, int64(meanEpochDays)*24*60*60, 12*60*60, 6)
```
Comment (eli-d) [inline cmd/microservice-epoch-rotation/main.go]: maybe move these numbers into a variable so it's easier to see what they are

File: cmd/microservice-epoch-rotation/main.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+	log.Debug(func(k *log.Log) {
+		k.Format("res: %v", gaussianRes)
+	})
```
Comment (eli-d) [inline cmd/microservice-epoch-rotation/main.go:197]: leftover debug?

File: cmd/microservice-epoch-rotation/main.go [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+		}
+	}
+
+	startTime := currentEpochEndTime
+
+	endTime := startTime.Add(closestDuration)
+
+	log.Debug(func(k *log.Log) {
+		k.Format("%v \n %v", startTime, endTime)
+	})
```
Comment (eli-d) [inline cmd/microservice-epoch-rotation/main.go:216]: leftover debug?

**Review by eli-d — APPROVED** [review]

**Pull request closed without merging**